src/workerd/api/sockets.h

```diff
@@ -26,7 +26,8 @@ class Socket: public jsg::Object {
 public:
   Socket(jsg::Lock& js, jsg::Ref<ReadableStream> readableParam, jsg::Ref<WritableStream> writable,
       kj::Own<jsg::PromiseResolverPair<void>> close, kj::Promise<void> connDisconnPromise,
-      jsg::Optional<SocketOptions> options)
+      jsg::Optional<SocketOptions> options, kj::TlsStarterCallback tlsStarter, bool isSecureSocket,
+      kj::String domain)
```

Comment:
Super minor nit since this function is local to sockets.c++ but I'm generally not a fan of bool arguments like this. Can the `kj::TlsStarterCallback` and `isSecureSocket` be combined into a `TlsOptions` struct?

```cpp
struct TlsOptions {
  kj::TlsStarterCallback tlsStarter;
  boolean secure = false;
};
```

Reply:
hmm, i'd rather keep it separate, these aren't really both Tls options. One is but the other is a socket state boolean.

Reply:
That's fine. In that case, however, I'd still prefer a bitwise flag or an enum class to a bool for `isSecureSocket`.